### DISCUSSION TITLE
Add more commands in background while program is running in foreground

### DISCUSSION THREAD

**Discussion by yonas** [nushell/nushell#14117] [Ideas]
For example, I came across these instructions:

```sh
cargo install perseus-cli
perseus new my-app
cd my-app/
perseus serve -w
```

I copy paste `cargo install perseus-cli` and hit Enter. Big mistake. It's taking forever to finish, and the last command - `perseus serve -w` - might take a long time as well.

It would be great to have a way to append the three commands after having entered the first.

**Comment by fdncred** [comment]
I'm not sure what "append commands" really means but I assume you mean having job management where you can launch commands in the bg and bring them to the fg. We want that too but it's pretty tricky cross platform. If you want to put them all on the same line, I believe you can do that and separate them with semi-colon.

**Comment by ghost** [comment]
You can paste multiple commands and then run them in sequence with a single `return`

If you want to run multiple commands at once (in the background) and know it before you start the first command you can use [pueue](https://github.com/Nukesor/pueue) ([nushell integration config snippet](https://github.com/Jan9103/nushell_pueue/blob/main/job.nu))

if you want to run multiple commands in sequence (in the background) and know it before you start the first command you can use [nq](https://github.com/leahneukirchen/nq)

if you don't expect to do more before pressing enter:
* many terminals(?) buffer all keypresses and then use them in order once something accepts them. this usually works and therefore you can often just paste the next commands. but i wouldnt rely on it and it can result in some unexpected behaviour if `apt` suddenly asks `are you sure? (y/n)`.  
* `crtl+z` would require background threads. the feature-request regarding that is [here](https://github.com/nushell/nushell/issues/247)
